### PR TITLE
Various cmake fixes

### PR DESCRIPTION
**Pull request by drwells**
As usual, the normal checklist isn't relevant.

The major fix here is that we need to specify to CMake that installed libraries should have the same rpath settings as the ones in the build directory - i.e., we don't support `LD_LIBRARY_PATH` ever.

@abarret This should fix the remaining issues for you - I installed Linux Mint 19 and used that to implement all the fixes. Could you please double-check?

### CODE REVIEW

**Comment by abarret** [comment]
This works for me.

File: cmake/IBAMRConfig.cmake.in [hunk 1]
```diff
@@ -19,20 +19,20 @@
 SET(MPI_ROOT "@MPI_ROOT@")
 FIND_PACKAGE(MPI REQUIRED)
 
-SET(Boost_FOUND "@Boost_FOUND@")
-SET(BOOST_ROOT "@BOOST_ROOT@")
-IF("${Boost_FOUND}")
+IF(NOT @IBAMR_USE_BUNDLED_Boost@)
+  SET(BOOST_ROOT "@BOOST_ROOT@")
```
Comment (abarret) [inline cmake/IBAMRConfig.cmake.in:23]: I think `Boost_ROOT` is used as the root.
```suggestion
  SET(BOOST_ROOT "@Boost_ROOT@")
```

Comment (drwells) [reply cmake/IBAMRConfig.cmake.in]: Yes, this should always be `Boost` - fixed.

**Comment by abarret** [comment]
This works with every build I have.

**Review by abarret — APPROVED** [review]

**Pull request merged**